src/test/java/com/exasol/adapter/dialects/sqlserver/SqlServerSqlDialectTest.java

```diff
@@ -139,6 +145,20 @@ void testApplyQuote() {
         assertThat(this.dialect.applyQuote("tableName"), equalTo("[tableName]"));
     }
 
+    @Test
+    void testApplyQuoteThrowsException() {
+        assertThrows(AssertionError.class, () -> this.dialect.applyQuote("[tableName]"));
+    }
```

Comment:
I think we need two separate test cases for names containing `[` and `]`. An additional test with a `"` would also make sense but I'm not sure whether that is even supported.